interpreter/src/objects.rs

```diff
@@ -416,115 +432,234 @@ impl Value {
+                        );
+                        (func)(&mut ctx)
                     }
-                    RelationOp::GreaterThanEq => {
-                        left.partial_cmp(&right)
-                            .ok_or(ExecutionError::ValuesNotComparable(left, right))?
-                            != Ordering::Less
+                    Some(target) => {
+                        let mut ctx = FunctionContext::new(
+                            call.func_name.clone().into(),
+                            Some(Value::resolve(target, ctx)?),
+                            ctx,
+                            call.args.clone(),
+                        );
+                        (func)(&mut ctx)
                     }
-                    RelationOp::Equals => right.eq(&left),
-                    RelationOp::NotEquals => right.ne(&left),
-                    RelationOp::In => match (left, right) {
-                        (Value::String(l), Value::String(r)) => r.contains(&*l),
-                        (any, Value::List(v)) => v.contains(&any),
-                        (any, Value::Map(m)) => match any.try_into() {
-                            Ok(key) => m.map.contains_key(&key),
-                            Err(_) => false,
-                        },
-                        (left, right) => Err(ExecutionError::ValuesNotComparable(left, right))?,
-                    },
-                };
-                Value::Bool(res).into()
-            }
-            Expression::Ternary(cond, left, right) => {
-                let cond = Value::resolve(cond, ctx)?;
-                if cond.to_bool() {
-                    Value::resolve(left, ctx)
-                } else {
-                    Value::resolve(right, ctx)
-                }
-            }
-            Expression::Or(left, right) => {
-                let left = Value::resolve(left, ctx)?;
-                if left.to_bool() {
-                    left.into()
-                } else {
-                    Value::resolve(right, ctx)
                 }
             }
-            Expression::And(left, right) => {
-                let left = Value::resolve(left, ctx)?;
-                if !left.to_bool() {
-                    Value::Bool(false).into()
+            Expr::Ident(name) => ctx.get_variable(name),
+            Expr::Select(select) => {
+                let left = Value::resolve(select.operand.deref(), ctx)?;
+                if select.test {
+                    match &left {
+                        Value::Map(map) => {
+                            for key in map.map.deref().keys() {
+                                if key.to_string().eq(&select.field) {
+                                    return Ok(Value::Bool(true));
+                                }
+                            }
+                            Ok(Value::Bool(false))
+                        }
+                        val => panic!("has() not supported on {val:?}"),
```

Comment:
This is not an `Err`, but should be 

```suggestion
                        _ => Ok(Value::Bool(false)),
```